### ISSUE TITLE
Mesh Collision / Navmesh Support / Rotatable Box Meshes

### ISSUE THREAD

**Issue by EGAMatsu** [deadcast2/UltraEd#137]
From the viewer, it seems rotating a Box Mesh just changes its size, which makes level design impossible, I assume it uses a AABB system though.

**Comment by deadcast2** [comment]
Yeah UltraEd is currently using AABB for efficiency. OOBB is very computationally expensive so I avoided implementing that. I will need some sort of way to add collision to imported level geometry like walls and floors. May have to use triangle based collision detection but I'll need to introduce space partioning for sure to not choke the CPU.

**Comment by Byterset** [comment]
Best case the colliders should not only be rotatable but can be completely manipulated on their own (rotate, resize, move). Basically like a child of the object.
I believe an additional capsule collider type would also be beneficial
